app/src/main/java/com/jerboa/ui/components/common/TextBadge.kt

```diff
@@ -33,12 +39,59 @@ fun TextBadge(
         Text(
             text = text,
             style = textStyle,
-            overflow = TextOverflow.Clip,
             maxLines = 1,
             color = textColor,
             modifier =
                 Modifier
-                    .padding(horizontalTextPadding.dp, verticalTextPadding.dp),
+                    .padding(horizontalTextPadding.dp, verticalTextPadding.dp)
+                    .basicMarquee(),
         )
     }
 }
+
+/**
+ * Displays activitypub items (communities, users), with a smaller @instance shown
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ItemAndInstanceTitle(
+    modifier: Modifier = Modifier,
+    title: String,
+    actorId: String?,
+    local: Boolean,
+    itemColor: Color = MaterialTheme.colorScheme.primary,
+    itemStyle: TextStyle = MaterialTheme.typography.labelLarge,
+    instanceColor: Color = MaterialTheme.colorScheme.onSurface.muted,
+    instanceStyle: TextStyle = MaterialTheme.typography.labelSmall,
+) {
+    val serverStr = if (!local) {
```

Comment:
I am not really fan of mixing ifs and their equivalents, But you can keep this if you want

Reply:
I fixed it up.